src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -140,101 +123,66 @@ EFeatureType RandomFourierGaussPreproc::get_feature_type() {
+void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth ) {
+	if (kernelwidth <= 0) {
+		error("InputError: kernelwidth must be greater 0!");
 	}
-	kernelwidth=kernelwidth2;
-}
 
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
+	is_updated = m_kernelwidth == kernelwidth;
+	SG_DEBUG("set_kernelwidth(), is_updated = {}",is_updated);
+	m_kernelwidth=kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
-
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
-
-	return false;
+float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
+	return (m_kernelwidth);
 }
 
-bool RandomFourierGaussPreproc::init_randomcoefficients() {
-	if (dim_feature_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_feature_space<=0 is not allowed\n");
+bool RandomFourierGaussPreproc::init_randomcoefficients(int32_t dim_input_space) {
+	if (m_dim_feature_space <= 0) {
+		error("ValueError: m_dim_feature_space has value <=0 which isn't allowed!, please set valid value via corresponding setter\n");
 	}
 	if (dim_input_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_input_space<=0 is not allowed\n");
+		error("InputError: dim_input_space must be greater 0!\n");
 	}
 
-	if (test_rfinited()) {
+	if (is_updated)
 		return false;
-	}
-
-
-	io::info("initializing randomcoefficients ");
-
-	float64_t pi = 3.14159265;
 
+	io::info("initializing randcoefficients ");
 
-	SG_FREE(randomcoeff_multiplicative);
-	randomcoeff_multiplicative=NULL;
-	SG_FREE(randomcoeff_additive);
-	randomcoeff_additive=NULL;
-
-
-	cur_dim_feature_space=dim_feature_space;
-	randomcoeff_additive=SG_MALLOC(float64_t, cur_dim_feature_space);
-	cur_dim_input_space = dim_input_space;
-	randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-
-	cur_kernelwidth=kernelwidth;
+	SG_FREE(m_randomweights);
+	m_randomweights=NULL;
+	SG_FREE(m_randomoffset);
+	m_randomoffset=NULL;
 
+	m_randomoffset= std::make_shared<SGVector<float64_t>>()
+	m_randomweights= 
+//TODO: Fill array and vector
 	random::fill_array(
-	    randomcoeff_additive, randomcoeff_additive + cur_dim_feature_space, 0.0,
-	    2 * pi, m_prng);
+	    m_randomoffset, m_randomoffset + m_dim_feature_space, 0.0,
+	    2 * M_PI, m_prng);
```

Comment:
where is M_PI coming from? Use the one in std::

Reply:
M_PI is from the cmath include, luckily it's already included it so it all blends in perfectly. We don't have std::pi. We can, however,  use ```const double pi = std::acos(-1);```  but that's a waste of LoC and even it's precision is lesser than M_PI:

Reply:
@karlnapf C++ library doesn't have pi definition, it relies on cmath. The best we could do is `static constexpr double PI = 3.1415...` or have an overloaded struct `shogun::constants<float>::pi`, `shogun::constants<double>::pi` and so on. @NanuSai avoid calculating constants inside an algorithm, make such things static and constexpr if possible

Reply:
Also calculating why calculate pi if it is a known constant?

Reply:
There seems to be a misunderstanding, I **already am** using M_PI. I was just explaining why I was using it.Anyway I guess we all agree on M_PI ;D.

Reply:
we want to drop that class eventually, but I guess you can use that for now.

@gf712 shall we put the constexpr in linalg maybe?